src/Symfony/Component/Lock/Store/StoreFactory.php

```diff
@@ -58,8 +58,19 @@ public static function createStore($connection)
                 return new FlockStore(substr($connection, 8));
             case 'semaphore' === $connection:
                 return new SemaphoreStore();
-            case class_exists(AbstractAdapter::class) && preg_match('#^[a-z]++://#', $connection):
-                return static::createStore(AbstractAdapter::createConnection($connection));
```

Comment:
I would prefer to keep this line and move the new logic there - that would allow both Cache and Lock to benefit from the new capability.

Reply:
Nevermind, after talking on Slack I realized this is not necessary.

Reply:
Do you mean adding Pdo DSN construction in cache `AbstractAdapter`?